Add instructions on how to run

daddydrac opened the pull request:
Please add step by step insttructions in a README.md file to get this going, so far its been a no go.

staticfloat committed:
Customizations

staticfloat committed:
Just mess it all up

staticfloat committed:
What was this `--renew-by-default` nonsense?  :P

staticfloat committed:
Slight wording change

staticfloat committed:
Version 0.5: Less features makes for better software

staticfloat committed:
More better error handling!

staticfloat committed:
Big rewrite so that we can enable sites as we get HTTPS certs

staticfloat committed:
Yet another rename

staticfloat committed:
Multitudinous fixes

staticfloat committed:
Cleanup fixes

staticfloat committed:
Let's wait for the NGINX pid only, so we don't wait on cron

staticfloat committed:
Cleanup `Dockerfile` a bit

staticfloat committed:
Add startup script support

staticfloat committed:
Better scripting support

staticfloat committed:
Get rid of useless `crontab` command

staticfloat committed:
Ditch `cron` for `bash`

staticfloat committed:
Add usage example to `README.md`

Commit:
add staging url for develop

Commit:
update datetime

valdergallo committed:
stop to reacreate keys on restart

valdergallo committed:
add old manteiner

valdergallo committed:
fix util sh

valdergallo committed:
add register script bash

valdergallo committed:
missing recreate the file for the next week

Commit:
sleep for more 10 sec

rodrigobraga committed:
add support to subdomains

rodrigobraga committed:
update version

rodrigobraga committed:
isolates parameter inside quotes

rodrigobraga committed:
fix scape

staticfloat committed:
Merge pull request #2 from rodrigobraga/master

Enable support to subdomains

gkosciolek committed:
Do not try to get certificate for wildcard domains as this is no longer supported by certbot

gkosciolek committed:
Fix the loop to make it sh compatible

staticfloat committed:
Merge pull request #3 from gkosciolek/master

Compatibility fixes

Commit:
set certbot as default server

staticfloat committed:
Merge pull request #4 from agairing/master

set certbot as default server

BrunoZell committed:
moved all source files to src folder

BrunoZell committed:
added baisc usage example

staticfloat committed:
Merge pull request #5 from BrunoZell/master

Added usage example

BrunoZell committed:
create src folder

BrunoZell committed:
merge valdergallo PR with master

BrunoZell committed:
remove register.sh

BrunoZell committed:
call certbot only once

BrunoZell committed:
add maintainer

BrunoZell committed:
move interval management to a util funcion

BrunoZell committed:
renaming

BrunoZell committed:
comments and documentation

BrunoZell committed:
fix function return

BrunoZell committed:
move more renewal logic to util.sh

BrunoZell committed:
put conditional renewal logic into certbot loop

BrunoZell committed:
comments

BrunoZell committed:
use domains private key file as timestamp

staticfloat committed:
Merge pull request #6 from BrunoZell/master

Fix problems with rate limit - updated

Commit:
Added IPv6 Support for the docker container
- Enabled IPv6 for the nginx server

staticfloat committed:
Merge pull request #8 from greendeath/master

Added IPv6 Support for the docker container

rachedhedia committed:
Update screen to use bash shell and adapt parse_keyfile output to remove starting $ character and trailing carriage return character : fixes certificate not found error

ds17f committed:
allow user configs without custom image

ds17f committed:
move link functionality into utils as a sh function

staticfloat committed:
Update screen to use bash shell and adapt parse_keyfile output to… (#10)

Update screen to use bash shell and adapt parse_keyfile output to rem…

ds17f committed:
address PR request comments for minor cleanup

chaopeng committed:
Update to acme v2

chaopeng committed:
Update to acme v2

staticfloat committed:
Merge pull request #13 from chaopeng/master

Update to acme v2

staticfloat committed:
Merge pull request #12 from ds17f/easy-user-config

Allow user configs without custom image

staticfloat committed:
Merge pull request #15 from yanokwa/opendatakit

Merge pull request #13 from chaopeng/master

ollien committed:
Avoid extracting comments when parsing domain names

staticfloat committed:
Merge pull request #16 from ollien/fix-comments

Avoid extracting comments when parsing domain names

staticfloat committed:
Enable templating!  Eliminate unnecessary customization!

lispmachine committed:
Stop container if nginx stops

lispmachine committed:
indentation fix

staticfloat committed:
Merge pull request #17 from lispmachine/wait-fix

Stop container if nginx stops

whatsmycoin committed:
Update docker-compose.yml

fix typo breaking sed in example

staticfloat committed:
Merge pull request #19 from whatsmycoin/patch-1

Update docker-compose.yml

staticfloat committed:
Protect against wedged `nginx` startups

If `nginx` starts up incorrectly, we'll end up waiting a week for it to
start up.  Instead, fail out first.

rigelk committed:
(minor) precision regarding ENVSUBST_VARS

staticfloat committed:
Use `ps | grep` instead of `jobs | grep`

It appears that `jobs` won't track the nginx processes started from a
config reload, so let's use `ps` instead.  Also, set `reuseport` on all
listen interfaces in order to dodge kernels that don't like `nginx`
restarting quickly.

staticfloat committed:
Exclude self

staticfloat committed:
Merge pull request #20 from rigelk/patch-1

(doc) precision regarding ENVSUBST_VARS

staticfloat committed:
Print out user configs as we template them

TrevorKS commented:
How do you put mupltiple configs and make it work? Can someone explain? I could only get it to work with one config file.  Adding a second for a second sever did not work

ollien committed:
Update README to indicate port 80 caeveat.

staticfloat committed:
Merge pull request #22 from ollien/patch-1

lnobach committed:
Switched to Python 3.7

staticfloat committed:
Merge pull request #23 from lnobach/master

staticfloat committed:
Tag v1.1

juehai committed:
added extra domains supported.

staticfloat committed:
Merge pull request #27 from juehai/master

added extra domains supported.

jippi committed:
Allow expanding certs via extra_domains_conf

Without this change, you get errors from certbot asking (interactively) if you want to replace/expand the certificate 

Manually fixing this involve deleting all the files for the domain in question and rerunning which is way more destructive

earthlingdavey committed:
Correct volume path for extra domains

Edit the container path for the `certbot_extra_domains` volume, to match the path in `scripts/utils.sh`.

staticfloat committed:
Merge pull request #39 from EarthlingDavey/master

Correct volume path for extra domains

staticfloat committed:
Merge pull request #35 from jippi/patch-1

Allow expanding certs via extra_domains_conf

armanjtehrani committed:
Update README.md

Fix nginx-template readme

jippi committed:
Update parse_domains parsing

This setup handles multiple vhosts with the same domain being processed multiple times during the "run_certbot" flow by finding all of them, sorting them, and taking only unique values for processing.

In my setup with more than 20 vhosts that have subdomains of a single root domain (and maintained via extra_domains configuration, this significantly limits the number of API hits to ACME, and prevents getting blocked when misconfigured

ledlamp committed:
"And a `.conf` in that directory" is misleading

Someone had a problem because they named their config file `.conf` which does NOT match the `*.conf` shell glob so it wasn't loaded.

staticfloat committed:
Merge pull request #43 from ledlamp/patch-1

"And a `.conf` in that directory" is misleading

staticfloat committed:
Merge pull request #40 from jooof/master

Update README.md

staticfloat committed:
Merge pull request #41 from jippi/patch-1

Update parse_domains parsing

staticfloat committed:
Update README.md
